xbmc/addons/AddonInstaller.cpp

```diff
@@ -189,22 +189,23 @@ bool CAddonInstaller::InstallModal(const std::string& addonID,
 
 bool CAddonInstaller::InstallOrUpdate(const std::string& addonID,
                                       BackgroundJob background,
-                                      ModalJob modal)
+                                      ModalJob modal,
+                                      int optionalTypeDependend /* = -1*/)
 {
   AddonPtr addon;
   RepositoryPtr repo;
   if (!CAddonInstallJob::GetAddon(addonID, repo, addon))
     return false;
 
   return DoInstall(addon, repo, background, modal, AutoUpdateJob::NO, DependencyJob::NO,
-                   AllowCheckForUpdates::YES);
+                   AllowCheckForUpdates::YES, optionalTypeDependend);
 }
 
 bool CAddonInstaller::InstallOrUpdateDependency(const ADDON::AddonPtr& dependsId,
                                                 const ADDON::RepositoryPtr& repo)
 {
   return DoInstall(dependsId, repo, BackgroundJob::NO, ModalJob::NO, AutoUpdateJob::NO,
-                   DependencyJob::YES, AllowCheckForUpdates::YES);
+                   DependencyJob::YES, AllowCheckForUpdates::YES, -1);
```

Comment:
Can this be changed to an enum (like the other arguments in this method) at least for consistency and avoid the magic number?

Reply:
Wouldn't be possible as directly as it is really a little magic number.
The associated add-on type must define its appropriate values.
I will use it as default with the:
```cpp
constexpr int ADDON_INSTALL_OPTIONAL_SELECT_UNDEFINED = -1;
```
Ah idea about a better name (where not so long, but still understandable)?